R/data.table.R

```diff
@@ -2758,9 +2758,14 @@ setcolorder = function(x, neworder=key(x), before=NULL, after=NULL, skip_absent=
   invisible(x)
 }
 
-set = function(x,i=NULL,j,value)  # low overhead, loopable
-{
-  .Call(Cassign,x,i,j,NULL,value)
+set = function(x, i=NULL, j, value) {
+  name = as.character(substitute(x))
```

Comment:
Doesn't this open the same can of worms that we're seeing with `setDT()`? #6706, #6723, ...

We need to consider `set(env$nm, ...)` and a whole litany of other ways `set()` can be used on `set(<call>, ...)` besides as `set(<name>, ...)`, right?